tests/e2e/tests/e2e/GitSelfSignCert.spec.ts

```diff
@@ -0,0 +1,106 @@
+ const gitPlugin: GitPlugin = e2eContainer.get(CLASSES.GitPlugin);
+ const testWorkspaceUtils: TestWorkspaceUtil = e2eContainer.get<TestWorkspaceUtil>(TYPES.WorkspaceUtil);
+ const dashboard: Dashboard = e2eContainer.get(CLASSES.Dashboard);
+ const browserTabsUtil: BrowserTabsUtil = e2eContainer.get(CLASSES.BrowserTabsUtil);
+ const editor: Editor = e2eContainer.get(CLASSES.Editor);
+
+ const workspaceName = 'gitSelfSignCert';
+
+
+ suite('Checking git + self sign cert', async () => {
+     const workspacePrefixUrl: string = `${TestConstants.TS_SELENIUM_BASE_URL}/dashboard/#/ide/${TestConstants.TS_SELENIUM_USERNAME}/`;
+     const gitProjectUrl = 'https://10.0.104.86:10080/maxura/gitService.git';
+     const committedFile = 'README.md';
+
+     suiteSetup(async function () {
+         const wsConfig = await testWorkspaceUtils.getBaseDevfile();
+         wsConfig.metadata!.name = workspaceName;
+         await browserTabsUtil.navigateTo(TestConstants.TS_SELENIUM_BASE_URL);
+         await loginPage.login();
+         await testWorkspaceUtils.createWsFromDevFile(wsConfig);
+     });
+
+     test('Login into workspace and open tree container', async () => {
+         await dashboard.openDashboard();
+         await dashboard.waitPage();
+         await browserTabsUtil.navigateTo(workspacePrefixUrl + workspaceName);
+         await ide.waitWorkspaceAndIde();
+         await projectTree.openProjectTreeContainer();
+         await driverHelper.wait(TimeoutConstants.TS_SELENIUM_LOAD_PAGE_TIMEOUT);
+     });
+
+
+     test('Clone project', async () => {
+         await cloneTestRepo(gitProjectUrl);
+
+         await projectTree.waitProjectImportedNoSubfolder('gitService');
+         await projectTree.waitItem('gitService/README.md');
+     });
+
+     test('Change commit and push', async function changeCommitAndPushFunc() {
+        const currentDate: string = Date.now().toString();
+        await projectTree.expandPathAndOpenFile('gitService', committedFile);
+        await editor.type(committedFile, currentDate + '\n', 1);
+        await gitPlugin.openGitPluginContainer();
+        await gitPlugin.waitChangedFileInChagesList(committedFile);
+        await gitPlugin.stageAllChanges(committedFile);
+        await gitPlugin.waitChangedFileInChagesList(committedFile);
+        await gitPlugin.typeCommitMessage(this.test!.title + currentDate);
+        await gitPlugin.commitFromCommandMenu();
+        await gitPlugin.pushChangesFromCommandMenu();
+        await quickOpenContainer.typeAndSelectSuggestion('git-admin', 'Git: https://10.0.104.86:10080 (Press \'Enter\' to confirm your input or \'Escape\' to cancel)');
+        await quickOpenContainer.typeAndSelectSuggestion('admin', 'Git: https://git-admin@10.0.104.86:10080 (Press \'Enter\' to confirm your input or \'Escape\' to cancel)');
+        await gitPlugin.waitDataIsSynchronized();
+        await driverHelper.wait(TimeoutConstants.TS_EDITOR_TAB_INTERACTION_TIMEOUT);
```

Comment:
Using the direct `wait` is not a good practice. Is there a reason for it as the next step is the stop of the workspace?

Reply:
I can comment this too.. This test is mostly based on tge 'GitSsh' E2E, and this wait is a compelled step.
Because the method which checks 'data synchronized' on UI waits disappearance specific event / locator, works properly, but these changes are not immediately appeared on GitHub side repo (may say not alwyas). So for reliable,  it was used this timeout.